packages/terra-list/src/MultiSelectList.jsx

```diff
@@ -0,0 +1,180 @@
+      return true;
+    }
+
+    for (let i = indexes.length - 1; i >= 0; i -= 1) {
+      if (this.state.selectedIndexes.indexOf(indexes[i]) >= 0) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  handleSelection(event, index) {
+    let newIndexes = [];
+    if (this.state.selectedIndexes.length) {
+      if (this.state.selectedIndexes.indexOf(index) >= 0) {
+        newIndexes = this.state.selectedIndexes.slice();
+        newIndexes.splice(newIndexes.indexOf(index), 1);
+      } else {
+        newIndexes = this.state.selectedIndexes.concat([index]);
+      }
+    } else {
+      newIndexes.push(index);
+    }
+
+    this.setState({ selectedIndexes: newIndexes });
+  }
+
+  shouldHandleSelection(index) {
+    if (this.state.selectedIndexes.length < this.validatedMaxCount()) {
+      return true;
+    }
+    if (this.state.selectedIndexes.indexOf(index) >= 0) {
+      return true;
+    }
+    return false;
+  }
+
+  cloneChildItems(items) {
+    const disableUnselectedItems = this.state.selectedIndexes.length >= this.validatedMaxCount();
+
+    return items.map((item, index) => {
+      const wrappedOnClick = this.wrappedOnClickForItem(item, index);
+      const newProps = this.newPropsForItem(item, index, wrappedOnClick, disableUnselectedItems);
+
+      return React.cloneElement(item, newProps);
+    });
+  }
+
+  wrappedOnClickForItem(item, index) {
+    const initialOnClick = item.props.onClick;
+    return (event) => {
+      if (this.shouldHandleSelection(index)) {
+        this.handleSelection(event, index);
+
+        if (this.onChange) {
+          this.onChange(event, this.state.selectedIndexes);
+        }
+      }
+
+      if (initialOnClick) {
+        initialOnClick(event);
+      }
+    };
+  }
+
+  newPropsForItem(item, index, onClick, disableUnselectedItems) {
+    const isSelected = this.state.selectedIndexes.indexOf(index) >= 0;
+
+    let newProps = { onClick };
+    if (isSelected !== item.isSelected) {
+      newProps.isSelected = isSelected;
+    }
+
+    if (item.props.isSelectable === undefined) {
+      newProps.isSelectable = true;
+    } else if (!item.props.isSelectable) {
+      newProps = {};
+    }
+
+    if (disableUnselectedItems && isSelected !== true) {
+      newProps.isSelectable = false;
+    }
+    return newProps;
+  }
+
+  validatedMaxCount() {
+    if (this.props.maxSelectionCount !== undefined) {
+      return this.props.maxSelectionCount;
+    }
+    return this.props.children.length;
+  }
+
+  unusedVariables(variable) {
+    return variable === this;
+  }
+
+  render() {
+    const { children, isDivided, onChange, maxSelectionCount, ...customProps } = this.props;
+    const clonedChildItems = this.cloneChildItems(children);
+
+    // Figure out how to handle this scenario.
+    this.unusedVariables(onChange);
+    this.unusedVariables(maxSelectionCount);
+
+    return (
+      <List isDivided={isDivided} {...customProps} tabindex="0">
```

Comment:
We need to camelCase tabindex
`tabindex` => `tabIndex`

In react, almost all hyphenated html attributes need to be camelcased: https://facebook.github.io/react/docs/dom-elements.html#all-supported-html-attributes